### PR TITLE
Implemented removal of reagents of zero coefficient

### PR DESCRIPTION
**Pull request by samseaver**
I tested this by artificially modifying a compound's formula.

### CODE REVIEW

File: Scripts/Biochemistry/Adjust_Reaction_Protons.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 Update_Reactions=0
 for rxn in sorted(Reactions_Dict.keys()):
+#    if(rxn != "rxn00013"):
+#        continue
+
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Adjust_Reaction_Protons.py:16]: Can we remove this?

File: Scripts/Biochemistry/Adjust_Reaction_Protons.py [hunk 2]
```diff
@@ -36,7 +39,9 @@
 
         old_stoichiometry=Reactions_Dict[rxn]["stoichiometry"]
         Rxn_Cpds_Array=ReactionsHelper.parseStoich(old_stoichiometry)
+        print "1: ",Rxn_Cpds_Array
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Adjust_Reaction_Protons.py:42]: Could be wordy. Is it needed?

File: Scripts/Biochemistry/Adjust_Reaction_Protons.py [hunk 3]
```diff
@@ -36,7 +39,9 @@
...
         ReactionsHelper.adjustCompound(Rxn_Cpds_Array,"cpd00067",float(number))
+        print "2: ",Rxn_Cpds_Array
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Adjust_Reaction_Protons.py:44]: Could be wordy. Is it needed?

File: Scripts/Biochemistry/Rebalance_Reactions.py [hunk 4]
```diff
@@ -11,22 +11,25 @@
 
 Update_Reactions=0
 for rxn in sorted(Reactions_Dict.keys()):
+#    if(rxn != "rxn00013"):
+#        continue
```
Comment (JamesJeffryes) [inline Scripts/Biochemistry/Rebalance_Reactions.py:15]: This code seems to duplicate logic in Adjust_Reaction_Protons.py. Can we consolidate it?

Comment (samseaver) [reply Scripts/Biochemistry/Rebalance_Reactions.py:15]: No, it started off like that, but it got tricky. There is an internal self-reference between status and stoichiometry when attempting to add/remove compounds, and it got more complex when considering whether users would add new compounds with an incorrectly stated number of protons. It was easier for me to separate it out into two scripts.

**Comment by samseaver** [comment]
Can you merge, I want to pull it into the other branch.

**Pull request merged**